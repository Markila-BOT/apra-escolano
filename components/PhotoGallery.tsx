import React, { FC, useState } from "react";
import ImgsViewer from "react-images-viewer";
import ReactPaginate from "react-paginate";
import { ITEMS_PER_PAGE } from "../constants";
import { Gallery } from "../interfaces";

type PhotoGalleryTypes = {
  gallery: Gallery[];
};

const PhotoGallery: FC<PhotoGalleryTypes> = ({ gallery }) => {
  const [isOpenImgViewer, setIsOpenImgViewer] = useState<boolean>(false);
  const [currentImageIndex, setCurrentImageIndex] = useState<number>(0);
  const [currentPage, setCurrentPage] = useState<number>(0);
  const offset = currentPage * ITEMS_PER_PAGE;
  const slicedGallery = gallery.slice(offset, offset + ITEMS_PER_PAGE);
  const pageCount = Math.ceil(gallery.length / ITEMS_PER_PAGE);
  const imagesSource = slicedGallery.map((gallery) => {
    return { src: gallery.url, caption: `Title: ${gallery.title}` };
  });

  const handlePageClick = ({ selected: selectedPage = 0 }) => {
    setCurrentPage(selectedPage);
  };

  return (
    <div className="flex flex-col mt-8">
      <div className="py-2 -my-2">
        <ImgsViewer
          imgs={imagesSource}
          isOpen={isOpenImgViewer}
          currImg={currentImageIndex}
          onClickPrev={() => setCurrentImageIndex((prev) => prev - 1)}
          onClickNext={() => setCurrentImageIndex((prev) => prev + 1)}
          onClose={() => setIsOpenImgViewer(false)}
        />
        <div className="inline-block min-w-full overflow-hidden align-middle border-b border-gray-200 shadow sm:rounded-lg">
          <table className="min-w-full">
            <thead>
              <tr>
                <th className="px-6 py-3 text-xs font-medium leading-4 tracking-wider text-left text-gray-500 uppercase border-b border-gray-200 bg-gray-50">
                  ID
                </th>
                <th className="px-6 py-3 text-xs font-medium leading-4 tracking-wider text-left text-gray-500 uppercase border-b border-gray-200 bg-gray-50">
                  Title
                </th>
                <th className="px-6 py-3 text-xs font-medium leading-4 tracking-wider text-left text-gray-500 uppercase border-b border-gray-200 bg-gray-50">
                  Thumbnail
                </th>
              </tr>
            </thead>
            <tbody className="bg-white">
              {slicedGallery.map((gallery, index) => (
                <tr key={gallery.id} className="odd:bg-white even:bg-slate-50">
                  <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
                    <div
                      className="text-sm font-medium leading-5 text-gray-900 cursor-pointer hover:text-blue-600"
                      onClick={() => {
                        setCurrentImageIndex(index);
                        setIsOpenImgViewer(true);
                      }}
                    >
                      {gallery.id}
                    </div>
                  </td>
                  <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
                    <div className="text-sm leading-5 text-gray-500">
                      {gallery.title}
                    </div>
                  </td>
                  <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
                    <div className="flex items-center justify-center">
                      <img
                        className="w-10 h-10 rounded-full cursor-pointer"
                        src={gallery.thumbnailUrl}
                        alt="gallery item"
                        onClick={() => {
                          setCurrentImageIndex(index);
                          setIsOpenImgViewer(true);
                        }}
                      />
                    </div>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
        <ReactPaginate
          previousLabel={"<<"}
          nextLabel={">>"}
          pageCount={pageCount}
          onPageChange={handlePageClick}
          containerClassName={"flex justify-end space-x-1 m-3"}
          pageClassName={
            "px-3 py-2 border border-gray-300 rounded hover:bg-gray-100"
          }
          activeClassName={
            "px-3 py-2 border border-blue-500 rounded bg-blue-500 text-white hover:bg-blue-500"
          }
          previousClassName={
            "px-3 py-2 border border-gray-300 rounded hover:bg-gray-100"
          }
          nextClassName={
            "px-3 py-2 border border-gray-300 rounded hover:bg-gray-100"
          }
        />
      </div>
    </div>
  );
};

export default PhotoGallery;
