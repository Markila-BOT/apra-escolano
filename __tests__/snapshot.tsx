import { render } from "@testing-library/react";
import PhotoGallery from "../components/PhotoGallery";
import SearchBar from "../components/SearchBar";

export const mockGallery = [
  {
    albumId: 1,
    id: 1,
    title: "mock1",
    url: "mock1",
    thumbnailUrl: "mock1",
  },
];

it("renders photog gallery unchanged", () => {
  const { container } = render(<PhotoGallery gallery={mockGallery} />);
  expect(container).toMatchSnapshot();
});

it("renders photog gallery unchanged", () => {
  const { container } = render(<SearchBar query="mock" setQuery={() => {}} />);
  expect(container).toMatchSnapshot();
});
