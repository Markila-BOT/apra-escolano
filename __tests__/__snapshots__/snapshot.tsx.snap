// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders photog gallery unchanged 1`] = `
<div>
  <div
    class="flex flex-col mt-8"
  >
    <div
      class="py-2 -my-2"
    >
      <div
        class="inline-block min-w-full overflow-hidden align-middle border-b border-gray-200 shadow sm:rounded-lg"
      >
        <table
          class="min-w-full"
        >
          <thead>
            <tr>
              <th
                class="px-6 py-3 text-xs font-medium leading-4 tracking-wider text-left text-gray-500 uppercase border-b border-gray-200 bg-gray-50"
              >
                ID
              </th>
              <th
                class="px-6 py-3 text-xs font-medium leading-4 tracking-wider text-left text-gray-500 uppercase border-b border-gray-200 bg-gray-50"
              >
                Title
              </th>
              <th
                class="px-6 py-3 text-xs font-medium leading-4 tracking-wider text-left text-gray-500 uppercase border-b border-gray-200 bg-gray-50"
              >
                Thumbnail
              </th>
            </tr>
          </thead>
          <tbody
            class="bg-white"
          >
            <tr
              class="odd:bg-white even:bg-slate-50"
            >
              <td
                class="px-6 py-4 whitespace-no-wrap border-b border-gray-200"
              >
                <div
                  class="text-sm font-medium leading-5 text-gray-900 cursor-pointer hover:text-blue-600"
                >
                  1
                </div>
              </td>
              <td
                class="px-6 py-4 whitespace-no-wrap border-b border-gray-200"
              >
                <div
                  class="text-sm leading-5 text-gray-500"
                >
                  mock1
                </div>
              </td>
              <td
                class="px-6 py-4 whitespace-no-wrap border-b border-gray-200"
              >
                <div
                  class="flex items-center justify-center"
                >
                  <img
                    alt="gallery item"
                    class="w-10 h-10 rounded-full cursor-pointer"
                    src="mock1"
                  />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <ul
        class="flex justify-end space-x-1 m-3"
      >
        <li
          class="px-3 py-2 border border-gray-300 rounded hover:bg-gray-100 disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            class=" "
            rel="prev"
            role="button"
            tabindex="-1"
          >
            &lt;&lt;
          </a>
        </li>
        <li
          class="px-3 py-2 border border-gray-300 rounded hover:bg-gray-100 px-3 py-2 border border-blue-500 rounded bg-blue-500 text-white hover:bg-blue-500"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            rel="canonical"
            role="button"
            tabindex="-1"
          >
            1
          </a>
        </li>
        <li
          class="px-3 py-2 border border-gray-300 rounded hover:bg-gray-100 disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Next page"
            class=" "
            rel="next"
            role="button"
            tabindex="-1"
          >
            &gt;&gt;
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`renders photog gallery unchanged 2`] = `
<div>
  <div
    class="flex items-center justify-center"
  >
    <div
      class="flex border-2 rounded"
    >
      <input
        autocomplete="off"
        class="px-4 py-2 w-80"
        id="search"
        name="search"
        placeholder="Search keywords on title"
        type="search"
        value="mock"
      />
      <div
        class="flex items-center justify-center px-4"
      >
        <svg
          class="w-6 h-6 text-gray-600"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.32 14.9l5.39 5.4a1 1 0 0 1-1.42 1.4l-5.38-5.38a8 8 0 1 1 1.41-1.41zM10 16a6 6 0 1 0 0-12 6 6 0 0 0 0 12z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
